.github/workflows/build.yml

```diff
@@ -0,0 +1,85 @@
+name: Build & Integration tests
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3.5.0
+        with:
+          distribution: 'temurin'
+          java-version: 11
+
+      - name: Setup gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Ktlint
+        run: ./gradlew lintKotlin --stacktrace
+
+      - name: Check Lint
+        run: ./gradlew lintDebug --stacktrace
+
+      - name: Build all build types
+        run: ./gradlew assemble --stacktrace
+
+      - name: Run Bucketeer unit tests
+        run: ./gradlew :bucketeer:testDebugUnitTest --stacktrace
+
+      - name: upload bucketeer build reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: bucketeer-build-reports
+          path: bucketeer/build/reports
+
+  androidTest:
+    needs: build
+    runs-on: macos-latest # enables hardware acceleration in the virtual machine
+    timeout-minutes: 45
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3.5.0
+        with:
+          distribution: 'temurin'
+          java-version: 11
+
+      - name: Setup gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: AVD cache
+        uses: actions/cache@v3
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-29
+      - name: Run instrumentation tests
+        uses: reactivecircus/android-emulator-runner@v2
```

Comment:
I'm going to switch to Gradle Managed Device once we update AGP to 7.3.x
GMD should be much more reliable.

https://developer.android.com/studio/test/gradle-managed-devices


https://github.com/android/sunflower/pull/785
https://github.com/android/nowinandroid/pull/210